pkg/leader/leader.go

```diff
@@ -43,16 +51,37 @@ func run(ctx context.Context, namespace, name string, client kubernetes.Interfac
 		logrus.Fatalf("error creating leader lock for %s: %v", name, err)
 	}
 
-	t := time.Second
-	if dl := os.Getenv("CATTLE_DEV_MODE"); dl != "" {
-		t = time.Hour
+	leaseDuration := defaultLeaseDuration
+	renewDeadline := defaultRenewDeadline
+	retryPeriod := defaultRetryPeriod
+	if d := os.Getenv("CATTLE_DEV_MODE"); d != "" {
+		leaseDuration = developmentLeaseDuration
+		renewDeadline = developmentRenewDeadline
+	}
+	if d := os.Getenv("CATTLE_ELECTION_LEASE_DURATION"); d != "" {
+		leaseDuration, err = time.ParseDuration(d)
+		if err != nil {
+			return fmt.Errorf("CATTLE_ELECTION_LEASE_DURATION value [%s] is not a valid duration: %w", d, err)
+		}
+	}
+	if d := os.Getenv("CATTLE_ELECTION_RENEW_DEADLINE"); d != "" {
+		renewDeadline, err = time.ParseDuration(d)
+		if err != nil {
+			return fmt.Errorf("CATTLE_ELECTION_RENEW_DEADLINE value [%s] is not a valid duration: %w", d, err)
+		}
+	}
+	if d := os.Getenv("CATTLE_ELECTION_RETRY_PERIOD"); d != "" {
+		retryPeriod, err = time.ParseDuration(d)
+		if err != nil {
+			return fmt.Errorf("CATTLE_ELECTION_RETRY_PERIOD value [%s] is not a valid duration: %w", d, err)
+		}
 	}
 
 	leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
```

Comment:
@olblak I was on paternity leave, so I was not watching updates.
@moio Our current guidelines are to add tests for all PRs, either Unit or Integration. Unit tests for this PR will have little immediate benefit but will help find regressions when someone goes to update this package in the future.
A good approach for adding tests would be to move most of this logic into a new function that returns a `leader.LeaderElectionConfig` and an `error` and then calls `leaderelection.RunOrDie(...)` with the output. This will allow future changes to the configuration to be easily tested as well.

We would get more mileage from creating Integration tests for this, but that is a much larger lift since there is no framework for it today.